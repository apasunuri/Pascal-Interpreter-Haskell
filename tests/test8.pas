program test8;

var
    a, g, b : real;

function power(num1, num2: real) : real;
var
    i, result: real;
begin
    if(num2 <= 0) then
        result := 1
    else
        result := num1 * power(num1, num2 - 1);
    power := result; 
end;

begin
    g := 22;
    b := 6;
    var
        b, c : real;
    for a := 8 to g do
        begin
            b := power(a, 4);
            writeln(b);
        end; 
    writeln(b);
end.