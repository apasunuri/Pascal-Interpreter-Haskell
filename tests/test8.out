4096.0
6561.0
10000.0
14641.0
20736.0
28561.0
38416.0
50625.0
65536.0
83521.0
104976.0
130321.0
160000.0
194481.0
234256.0
6.0